src/test/java/org/flickit/flickitassessmentcore/application/service/answer/SubmitAnswerServiceTest.java

```diff
@@ -0,0 +1,154 @@
+package org.flickit.flickitassessmentcore.application.service.answer;
+
+import org.flickit.flickitassessmentcore.application.port.in.answer.SubmitAnswerCommand;
+import org.flickit.flickitassessmentcore.application.port.out.answer.CheckAnswerExistenceByAssessmentResultIdAndQuestionIdPort;
+import org.flickit.flickitassessmentcore.application.port.out.answer.LoadAnswerByAssessmentResultIdAndQuestionIdPort;
+import org.flickit.flickitassessmentcore.application.port.out.answer.SaveAnswerPort;
+import org.flickit.flickitassessmentcore.application.port.out.answer.UpdateAnswerPort;
+import org.flickit.flickitassessmentcore.application.port.out.assessmentresult.CheckAssessmentResultExistencePort;
+import org.flickit.flickitassessmentcore.application.port.out.assessmentresult.InvalidateAssessmentResultPort;
+import org.flickit.flickitassessmentcore.application.service.exception.ResourceNotFoundException;
+import org.flickit.flickitassessmentcore.domain.Answer;
+import org.flickit.flickitassessmentcore.domain.AssessmentResult;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Spy;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.util.UUID;
+
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.Mockito.*;
+
+@ExtendWith(MockitoExtension.class)
+class SubmitAnswerServiceTest {
+
+    @Spy
+    @InjectMocks
+    private SubmitAnswerService service;
+
+    @Mock
+    private SaveAnswerPort saveAnswerPort;
+
+    @Mock
+    private UpdateAnswerPort updateAnswerPort;
+
+    @Mock
+    private LoadAnswerByAssessmentResultIdAndQuestionIdPort loadAnswerPort;
+
+    @Mock
+    private InvalidateAssessmentResultPort invalidateAssessmentResultPort;
+
+    @Mock
+    private CheckAssessmentResultExistencePort assessmentResultExistencePort;
+
+    @Mock
+    private CheckAnswerExistenceByAssessmentResultIdAndQuestionIdPort checkAnswerExistencePort;
+
+    @Test
+    void submitAnswer_SavesAnswerAndInvalidatesAssessmentResult_WhenAnswerNotExist() {
+        SubmitAnswerCommand command = new SubmitAnswerCommand(
+            UUID.randomUUID(),
+            1L,
+            1L
+        );
+        doReturn(true).when(assessmentResultExistencePort).existsById(command.getAssessmentResultId());
```

Comment:
طبق مستندات Mockito توصیه می‌شه از به صورت when thenReturn استفاده بشه. فقط در شرایطی که استفاده ازش امکان‌پذیر نیست، از doReturn when استفاده بشه. جاواداک متد doReturn:
```java
/**
     * Use <code>doReturn()</code> in those rare occasions when you cannot use {@link Mockito#when(Object)}.
     * <p>
     * <b>Beware that {@link Mockito#when(Object)} is always recommended for stubbing because it is argument type-safe
     * and more readable</b> (especially when stubbing consecutive calls).

```
لطفا تست‌ها را با توجه به این نکته اصلاح کنید.

Reply:
انجام شد